uefi/src/proto/device_path/mod.rs

```diff
@@ -326,6 +356,75 @@ impl DevicePath {
             stop_condition: StopCondition::EndEntireNode,
         }
     }
+
+    #[cfg(feature = "alloc")]
+    pub fn to_text_by_node(
+        &self,
+        bs: &BootServices,
+    ) -> crate::Result<Vec<(&DevicePathNode, CString16)>> {
+        use uefi::Identify;
+        let &handle = bs
+            .locate_handle_buffer(SearchType::ByProtocol(&DevicePathToText::GUID))
+            .unwrap()
+            .first()
+            .unwrap();
+
+        let to_text_protocol = unsafe {
+            bs.open_protocol::<DevicePathToText>(
```

Comment:
Do the corresponding `close_protocol` calls happen via `Drop`? 

Also I assume you'll still build the error propagation?

Reply:
yes, this is just a q&d approach to get early feedback

by the way, this MR has some commits in common with #826 - in case you wonder